src/hotspot/cpu/s390/stubGenerator_s390.cpp

```diff
@@ -3053,6 +3053,31 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  // load Method* target of MethodHandle
+  // Z_ARG1 = jobject receiver
+  // Z_method = Method* result
+  address generate_upcall_stub_load_target() {
+    StubCodeMark mark(this, "StubRoutines", "upcall_stub_load_target");
+    address start = __ pc();
+    __ save_return_pc();
```

Comment:
@offamitkumar: Is saving and restoring the return_pc needed? Isn't in preserved by load_heap_oop?

Reply:
Tier1 test are fine with/with "saving & restoring" return_pc;

Reply:
I found it: https://github.com/openjdk/jdk/blob/433f6d8a0643b59663bf76c0f3a2af27a6cc56b7/src/hotspot/cpu/s390/gc/g1/g1BarrierSetAssembler_s390.cpp#L238
Other GCs with load barriers are not implemented, so the save&restore code is redundant.
The stub is frameless and only needs the save&restore code when calling C.

Reply:
@JornVernee would you please remove "saving & restoring" code for the return PC as mentioned by Martin. Thanks.